chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UserDefinedTriggersEmailJob.java

```diff
@@ -0,0 +1,251 @@
+        row.add("");
+        return row;
+    }
+
+    private void sendEmail(JobExecutionContext jobContext, List<List<String>> csvRows)
+            throws MessagingException {
+        LOGGER.info("Sending email to {} with contents {} and a total of {} user triggers",
+                getEmailRecipients(jobContext).get(0), getHtmlMessage(csvRows.size()));
+
+        EmailBuilder emailBuilder = new EmailBuilder(env);
+        emailBuilder.recipients(getEmailRecipients(jobContext))
+        .subject(getSubject(jobContext))
+        .htmlMessage(getHtmlMessage(csvRows.size()))
+        .fileAttachments(getAttachments(csvRows))
+        .acbAtlHtmlFooter()
+        .sendEmail();
+    }
+
+    private String getSubject(JobExecutionContext jobContext) {
+        return env.getProperty("userTriggersReport.subject");
+    }
+
+    private List<File> getAttachments(List<List<String>> csvRows) {
+        List<File> attachments = new ArrayList<File>();
+        File csvFile = getCsvFile(csvRows);
+        if (csvFile != null) {
+            attachments.add(csvFile);
+        }
+        return attachments;
+    }
+
+    private File getCsvFile(List<List<String>> csvRows) {
+        File csvFile = null;
+        if (csvRows.size() > 0) {
+            String filename = env.getProperty("userTriggersReport.filename");
+            if (csvRows.size() > 0) {
+                csvFile = getOutputFile(csvRows, filename);
+            }
+        }
+        return csvFile;
+    }
+
+    private File getOutputFile(List<List<String>> rows, String reportFilename) {
+        File temp = null;
+        try {
+            temp = File.createTempFile(reportFilename, ".csv");
+            temp.deleteOnExit();
+
+            try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(temp),
+                    Charset.forName("UTF-8").newEncoder());
+                    CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.EXCEL)) {
+                writer.write('\ufeff');
+                csvPrinter.printRecord(getHeaderRow());
+                for (List<String> rowValue : rows) {
+                    csvPrinter.printRecord(rowValue);
+                }
+                csvPrinter.printRecord(getFooterRow());
+            } catch (IOException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return temp;
+    }
+
+    private String getHtmlMessage(Integer rowCount) {
+        return String.format(env.getProperty("userTriggersReport.htmlBody"), rowCount);
+    }
+
+    private List<String> getEmailRecipients(JobExecutionContext jobContext) {
+        return Arrays.asList(jobContext.getMergedJobDataMap().getString("email"));
+    }
+
+    private void setSecurityContext() {
+        JWTAuthenticatedUser adminUser = new JWTAuthenticatedUser();
+        adminUser.setFullName("Administrator");
+        adminUser.setId(-2L);
```

Comment:
I think @tmy1313 called me out on not using some sort of constant like `User.ADMIN_USER_ID` from the `gov.healthit.chpl.auth.user;` package before